flake.nix

```diff
@@ -0,0 +1,79 @@
+{
+  description = "Slack library for Haskell";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  nixConfig.allow-import-from-derivation = true; # cabal2nix uses IFD
+
+  outputs = { self, nixpkgs, flake-utils }:
+    let
+      ghcVer = "ghc924";
+      makeHaskellOverlay = overlay: final: prev: {
+        haskell = prev.haskell // {
+          packages = prev.haskell.packages // {
+            ${ghcVer} = prev.haskell.packages."${ghcVer}".override (oldArgs: {
+              overrides =
+                prev.lib.composeExtensions (oldArgs.overrides or (_: _: { }))
+                  (overlay prev);
+            });
+          };
+        };
+      };
+
+      out = system:
+        let
+          pkgs = import nixpkgs {
+            inherit system;
+            overlays = [ self.overlays.default ];
+            config.allowBroken = true;
+          };
+
+        in
+        {
+          packages = rec {
+            default = slack-web;
+            slack-web = pkgs.haskell.packages.${ghcVer}.slack-web;
+          };
+
+          checks = {
+            inherit (self.packages.${system}) slack-web;
+          };
+
+          # for debugging
+          # inherit pkgs;
+
+          devShells.default =
+            let haskellPackages = pkgs.haskell.packages.${ghcVer};
+            in
+            haskellPackages.shellFor {
+              packages = p: [ self.packages.${system}.slack-web ];
+              withHoogle = true;
+              buildInputs = with haskellPackages; [
+                haskell-language-server
+                fourmolu
+                # jacked on this particular nixpkgs version
+                # ghcid
```

Comment:
Oh, no! D-:

Reply:
I'm going to run a bunch of compile and we'll see if a newer nixpkgs version fixes it. Unfortunately there's no caching for ghc924 on macOS/aarch64.

Reply:
hahahaha lol, fourmolu is jacked on new nixpkgs. I know why, however.